src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -471,31 +478,30 @@ class MariaDBManager implements IQueryService, INotificationService {
     return res
   }
 
-  private Person getPersonNameByID(int id) {
-    //    logger.info("Looking for user with email " + email + " in the DB");
-    Person res = null;
+  private static Person rowResultToPerson(GroovyRowResult rowResult) {
+    String firstName = rowResult.get("first_name")
+    String lastName = rowResult.get("last_name")
+    String email = rowResult.get("email")
+    if (firstName == null || lastName == null || email == null) {
+      log.error("Incomplete person dataset. ${rowResult.get("id")}")
+      throw new RuntimeException("Incomplete person dataset.")
+    }
+    return new Person(firstName, lastName, email)
+  }
+
+  private Person getPersonByDbId(int id) {
     Connection connection = Objects.requireNonNull(dataSource.getConnection(), "Connection must " +
             "not be null.")
-    Sql sql = new Sql(connection)
-
-    final String query = "SELECT * from $PERSONS_TABLE WHERE id = ${id};"
-    try {
-      List<GroovyRowResult> results = sql.rows(query)
-      if( results.size() > 0 ) {
-        GroovyRowResult rs = results.get(0)
-        //        logger.info("email found!");
-        String firstName = rs.get("first_name")
-        String lastName = rs.get("last_name")
-        String email = rs.get("email")
-        res = new Person(firstName, lastName, email)
-      }
-    } catch (SQLException e) {
-      //      logger.error("SQL operation unsuccessful: " + e.getMessage());
-      e.printStackTrace();
-    } finally {
-      sql.close()
+    String query = "SELECT * from ${PERSONS_TABLE} WHERE id = ${id}"
+    try (Sql sql = new Sql(connection)) {
+      def results = sql.rows(query)
+      Person foundPerson = Optional.ofNullable(results)
+              .filter({ return !(it == null || it.isEmpty()) })
+              .map(DefaultGroovyMethods::first)
+              .map(this::rowResultToPerson)
+              .orElseThrow({ new NotFoundException("No person with id ${id} exists.")})
+      return foundPerson
```

Comment:
where did the catch clause go? Why did you delete it?

Reply:
I would argue to remove the try clause when doing it this way

Reply:
This is a try with resources https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html making sure that the SQL is closed correctly.

Reply:
Ok cool, thanks for the link!